app/models/task.rb

```diff
@@ -35,7 +35,10 @@ class Task < ApplicationRecord
 
   scope :owner, ->(user) { where(user:) }
   scope :public_access, -> { where(access_level: 'public') }
-  scope :visibility, ->(visibility, user = nil) { {owner: owner(user), public: public_access}.with_indifferent_access[visibility] }
+  scope :private_access, ->(user) { where(access_level: 'private', user:) }
```

Comment:
This private access shows only those tasks the given `user` is the author _and_ those who are `private`. Depending on the use case, this is okay / desired, but might change depending on the discussion in https://github.com/openHPI/codeharbor/pull/905#discussion_r1211467594.